src/util/strencodings.cpp

```diff
@@ -526,3 +527,46 @@ std::string HexStr(const Span<const uint8_t> s)
     assert(it == rv.end());
     return rv;
 }
+
+std::optional<uint64_t> ParseByteUnits(const std::string& str, uint64_t default_multiplier)
+{
+    if (str.empty()) {
+        return std::nullopt;
+    }
+    assert(default_multiplier > 0);
+    auto multiplier = default_multiplier;
+    char unit = str.back();
+    switch (unit) {
+    case 'k':
+        multiplier = 1000;
+        break;
+    case 'K':
+        multiplier = 1024;
+        break;
+    case 'm':
+        multiplier = 1000 * 1000;
+        break;
+    case 'M':
+        multiplier = 1ULL << 20;
+        break;
+    case 'g':
+        multiplier = 1000ULL * 1000 * 1000;
```

Comment:
```suggestion
        multiplier = 1000'000'000ULL;
```

Haven't tried this, but can the compiler parse this?

Reply:
it does!